app/models/proxy.rb

```diff
@@ -511,6 +515,23 @@ def service_mesh_integration?
     Service::DeploymentOption.service_mesh.include?(deployment_option)
   end
 
+  def find_or_create_proxy_config_affecting_change
+    proxy_config_affecting_change || create_proxy_config_affecting_change
+  end
+  alias affecting_change_history find_or_create_proxy_config_affecting_change
+  private :find_or_create_proxy_config_affecting_change
+
+  def pending_affecting_changes?(environment: :sandbox)
+    return unless apicast_configuration_driven?
+    config = latest_config(environment: environment)
+    return false unless config
+    config.created_at < affecting_change_history.updated_at
+  end
+
+  def latest_config(environment: :sandbox)
```

Comment:
I think we won't need it for production, as production is always a promotion from sandbox
Diff between production and sandbox can already be done by checking the ProxyConfig